### ISSUE TITLE
outputRelabeling may alter and contaminate the shared slice memory for labels in streamAggr

### ISSUE THREAD

**Issue by ChengChung** [VictoriaMetrics/VictoriaMetrics#5402]
### Describe the bug

```
type flushCtx struct {
	skipAggrSuffix bool
	suffix         string

	tss     []prompbmarshal.TimeSeries
	labels  []prompbmarshal.Label
	samples []prompbmarshal.Sample
}
```

The struct `streamaggr.flushCtx` optimizes memory usage by reusing `flushCtx.labels` in both `flushCtx.reset()` and `flushCtx.appendSeries()`, thereby minimizing memory reallocations.

In `flushCtx.tss`, the `labels` of timeseries data are references to a partial slice of `flushCtx.labels`. It is permissible to modify the values of existing labels without directly resizing or appending new labels to the existing timeseries.

However, a potential issue arises when `aggregator.flush()` invokes the function `aggrState.appendSeriesForFlush()`, which subsequently calls `outputRelabeling.Apply()`. This call directly modifies `ts.Labels` in place, allowing for the addition of new labels. This action has the unintended consequence of affecting labels out of bounds and subsequently impacting the labels of the next timeseries, as they share the same slice.

For instance, appending a new label, such as `vmaggr="test2"` in the outputRelabel section, triggers the `parsedRelabelConfig.apply()` function. Within this function, the `Fast path for the rule that sets label value` (found [here](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/2ec2deebb2139de44ffa021c93f9b028464ea5d9/lib/promrelabel/relabel.go#L214)) appends a new label to the input argument `labels` when the label `vmaggr` is not found in the argument `labels`. This behavior can lead to modifying labels beyond the bounds of the current timeseries, affecting subsequent timeseries sharing the same slice.

![vmagent-draw](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/9165331/986e96bf-dbf5-4d70-89cf-f922bd80abf8)

### To Reproduce

```
- match: 'node_network_transmit_bytes_total{job="node"}'
  interval: 5m
  by: [hostname]
  outputs: [total]
  output_relabel_configs:
  - target_label: vmaggr
    replacement: test2
```

In the context of the provided configuration for the VMagent, it is crucial to ensure the generation of more than one time series after streamAggr. Specifically, having more than one `hostname` is necessary to reproduce the observed bug. Notably, the target label `vmaggr` is not present as part of the labels in the aggregated series prior to output relabeling.

Following the execution of streamAggr, certain datapoints are found to be missing or incorrect, as demonstrated by the following results:

```
curl http://example.com/api/v1/export -X POST -d 'match[]={hostname="gd-rpi", vmaggr="test2"}&start=1701135959&end=1800000000' |jq .
```

The resulting data includes missing datapoints and inaccuracies in the node_network_transmit_bytes_total:5m_by_hostname_total metric:

```
{
  "metric": {
    "__name__": "node_network_transmit_bytes_total:5m_by_hostname_total",
    "hostname": "gd-rpi",
    "vmaggr": "test2"
  },
  "values": [
    32999759689,
    33098974934.001945,
    33405793344,
    33522280196,
    33623253759,
    33859721508.001965,
    33973027935,
    34075887544.001953,
    34186708294,
    34302568163.001953,
    34397013776,
    34608992766,
    34746086722,
    35095700130,
    35333894376,
    35989502473,
    36117837484
  ],
  "timestamps": [
    1701135959000,
    1701136259000,
    1701137159000,
    1701137459000,
    1701137759000,
    1701138359000,
    1701138659000,
    1701138959000,
    1701139259000,
    1701139559000,
    1701139859000,
    1701140459000,
    1701140759000,
    1701141659000,
    1701142259000,
    1701144059000,
    1701144359000
  ]
}
{
  "metric": {
    "hostname": "gd-rpi",
    "vmaggr": "test2"
  },
  "values": [
    33207895927,
    33302643823,
    33747304413,
    34501648391,
    34865306261.00195,
    34989989092,
    35210742636,
    35449621473,
    35558411921,
    35662837511,
    35768706125,
    35880406414,
    36256835814.99805
  ],
  "timestamps": [
    1701136559000,
    1701136859000,
    1701138059000,
    1701140159000,
    1701141059000,
    1701141359000,
    1701141959000,
    1701142559000,
    1701142859000,
    1701143159000,
    1701143459000,
    1701143759000,
    1701144659000
  ]
}
```

We notice missing `node_network_transmit_bytes_total:5m_by_hostname_total{hostname="gd-rpi", vmaggr="test2"}` datapoints and another metric lacking the `__name__` fields. However, combining both datasets results in a complete set of information.

### Version

vmagent-20231127-082647-tags-v1.93.8-cluster-0-g2ec2deebb-dirty-707d986c

### Used command-line flags

-promscrape.maxScrapeSize=1024000000 -remoteWrite.queues=24 -remoteWrite.tmpDataPath=/var/lib/vmagent-remotewrite-data -promscrape.config=/etc/vmagent/prometheus.yml -remoteWrite.streamAggr.config=/etc/vmagent/aggr.yml -remoteWrite.streamAggr.keepInput=true -remoteWrite.url=http://example.com/api/v1/write -promscrape.configCheckInterval=30s

### Additional information

Here is a straightforward code example that demonstrates the behavior of a shared slice

```
package main

import "fmt"

func main() {
	labels := make([]string, 0, 100)

	offset := 0
	labels = append(labels, "label1")
	labels = append(labels, "label2")
	labels = append(labels, "label3")

	a := labels[offset:]
	offset += len(a)

	labels = append(labels, "label4")
	labels = append(labels, "label5")
	labels = append(labels, "label6")
	b := labels[offset:]
	offset += len(b)

	fmt.Println(a, b)
	fmt.Println(labels)

	a = append(a, "label7")
	fmt.Println(a, b)
	fmt.Println(labels)
}

```

the output is 
```
[label1 label2 label3] [label4 label5 label6]
[label1 label2 label3 label4 label5 label6]
[label1 label2 label3 label7] [label7 label5 label6]
[label1 label2 label3 label7 label5 label6]
```

appending items to 'a' directly influences the content of 'b' due to their shared underlying array

**Comment by f41gh7** [comment]
Thanks for the detailed report. We're going to fix it as soon as possible.

**Comment by valyala** [comment]
The bug should be fixed in the commit 41f7940f973404a7b93f446f2f9055842d72c103 . This commit will be included in the next release. In the mean time it is possible to verify the bugfix by running `vmagent` built from this commit according to [these docs](https://docs.victoriametrics.com/vmagent.html#how-to-build-from-sources).

**Comment by valyala** [comment]
FYI, the bugfix has been included in [v1.93.9 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.93.9). This bugfix will be also included in the latest release soon.

**Comment by valyala** [comment]
The bugfix has been included in [vmagent v1.96.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue as fixed.
